src/Build/BackEnd/Components/SdkResolution/SdkResult.cs

```diff
@@ -33,30 +33,61 @@ public SdkResult(SdkReference sdkReference, string path, string version, IEnumer
         {
             Success = true;
             SdkReference = sdkReference;
-            _path = path;
-            _version = version;
+            Path = path;
+            Version = version;
             Warnings = warnings;
         }
 
         public SdkResult()
         {
+
         }
 
-        public Construction.ElementLocation ElementLocation { get; set; }
+        public SdkResult(SdkReference sdkReference, IEnumerable<SdkResultPathAndVersion> paths, IDictionary<string, string> propertiesToAdd,
+                         IDictionary<string, SdkResultItem> itemsToAdd, IEnumerable<string> warnings)
+        {
+            Success = true;
+            SdkReference = sdkReference;
+            if (paths != null)
+            {
+                var firstPathAndVersion = paths.FirstOrDefault();
+                if (firstPathAndVersion != null)
+                {
+                    Path = firstPathAndVersion.Path;
+                    Version = firstPathAndVersion.Version;
+                }
```

Comment:
Can you explain why this pattern rather than allowing Path to be null if multiple-paths are defined?

Reply:
Are you asking why if there are multiple paths, it doesn't just use a List to hold them and leave singular `Path` property set to null?

It's fairly arbitrary.  The SdkResult should always be created with the Factory method that takes a list of paths, so the way the multiple paths are stored isn't really exposed outside of MSBuild.